components/core/src/clp/CurlDownloadHandler.hpp

```diff
@@ -63,7 +65,9 @@ class CurlDownloadHandler {
             size_t offset = 0,
             bool disable_caching = false,
             std::chrono::seconds connection_timeout = cDefaultConnectionTimeout,
-            std::chrono::seconds overall_timeout = cDefaultOverallTimeout
+            std::chrono::seconds overall_timeout = cDefaultOverallTimeout,
+            std::map<std::string, std::string> const& custom_headers
+            = std::map<std::string, std::string>()
```

Comment:
Similar to the comment in `NetworkReader`. Notice that here we prefer to pass a const reference because there's no need to transfer the ownership of the header map

Reply:
fixed in next commit